proxylib/memcached/binary/parser.go

```diff
@@ -0,0 +1,369 @@
+// Copyright 2018 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package binary
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/cilium/cilium/pkg/envoy/cilium"
+	"github.com/cilium/cilium/proxylib/proxylib"
+
+	log "github.com/sirupsen/logrus"
+)
+
+// BinaryMemcacheRule matches against memcached requests
+type BinaryMemcacheRule struct {
```

Comment:
Will the policy rule be the same for the binary and text based protocol?

Reply:
@tgraf  From user perspective yes, but `Matches` implementation will be different for them. I am checking while implementing text protocol how can I make these 2 types of rules behave the same from user side and differently on protocol side.

Reply:
OK. The interface is the string map anyway so we are flexible. I think it will make sense to move some of the common code into `proxylib/memcache/`